python/tidy/servo_tidy_tests/test_tidy.py

```diff
@@ -30,6 +32,21 @@ def test_tidy_config(self):
         self.assertEqual('invalid config table [wrong]', errors.next()[2])
         self.assertNoMoreErrors(errors)
 
+    def test_directory_checks(self):
+        mock_config = {
+            "check_ext": {
+                "./python/tidy/servo_tidy_tests/dir_check/webidl_plus/": ['webidl', 'test'],
+                "./python/tidy/servo_tidy_tests/dir_check/only_webidl/": ['webidl'],
+                "./python/tidy/servo_tidy_tests/dir_check/empty/": ['webidl']
+            }
+        }
+        with mock.patch.dict('servo_tidy.tidy.config', mock_config):
+            errors = tidy.check_directory_files(False)
+            error_dir = "./python/tidy/servo_tidy_tests/dir_check/webidl_plus"
+            self.assertEqual("Unexpected extension found for test.rs. We only expect webidl, test in {0}".format(error_dir), errors.next()[2])
+            self.assertEqual("Unexpected extension found for test2.rs. We only expect webidl, test in {0}".format(error_dir), errors.next()[2])
+            self.assertNoMoreErrors(errors)
```

Comment:
It's a good idea, and the problem is that `collect_errors_for_files` is really written for the 'standard checks' part, which checks individual files. It would require changing quite a few things to plug this "directory checks" part into it.

For example:

* In the main `scan` function, `collect_errors_for_files` only receives files that have some limited subset of expected extensions, while in our case we actually want to check for the absence of any type of unexpected files. 

* We could call `collect_errors_for_files` separately just for our part of the workflow, however `collect_errors_for_files` contains all sorts of stuff that makes it unusable in our case, such as [specific errors for empty files](https://github.com/servo/servo/blob/master/python/tidy/servo_tidy/tidy.py#L746) and [printing an inappropriate message in our case](https://github.com/servo/servo/blob/master/python/tidy/servo_tidy/tidy.py#L738).

Right now we are already re-using the `get_file_list`, which is more generic, and I feel like the `collect_errors_for_files` is not generic, rather written specificaly for the other checks. 

I'd say it's easier to not try to re-use it, and keep the "directory checks" separate from the other "file checks". 


For example, that part of the workflow only checks files with certain extensions, while in this case we really want to check for the presence of any type of unexpected files.  There is also some other small stuff, for example this line: